agent/agent.go

```diff
@@ -91,35 +91,36 @@ func (a *Agent) Run(payload *queue.Work, cancel <-chan bool) error {
 }
 
 func (a *Agent) prep(w *queue.Work) (*yaml.Config, error) {
-
 	envs := toEnv(w)
 	w.Yaml = expander.ExpandString(w.Yaml, envs)
 
-	// inject the netrc file into the clone plugin if the repository is
-	// private and requires authentication.
+	// append all secrets. they will be checked by the ImageSecrets transform.
 	var secrets []*model.Secret
-	if w.Verified {
-		secrets = append(secrets, w.Secrets...)
-	}
+	secrets = append(secrets, w.Secrets...)
```

Comment:
> if you define a global s3 with no verify, and a repo specific s3 with verify, and a build doesn't have a verified yaml, it will take the global one even if the credentials are not working for the project

I think this is the desired behavior. If you setup a global secret with no verification, you should always expect it to be used without verification.

In general I would personally never use global secrets. They are a poor security choice, but I understand how they can be useful for private on-premise installations. I don't think we need to overcomplicate the implementation to provide added security to something that is always going to be insecure.